### DISCUSSION TITLE
frontmatter not picking up the value of the defined field.

### DISCUSSION THREAD

**Discussion by 82Kang** [SilentVoid13/Templater#763] [Help]
I am trying the templater example for frontmatter from the [documentation site](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/frontmatter-module.html).

alias: myfile
note type: seedling
my type: <% tp.frontmatter.alias %>

When I apply this template in a note, I am getting:

alias: myfile
note type: seedling
my type: undefined

And if I re-apply the template to the same note, I am getting the desired results:

alias: myfile
note type: seedling
my type: undefined---
alias: myfile
note type: seedling
my type: myfile

Is it a bug or I am not using frontmatter the right way?

**Comment by welpdx** [comment]
The documentation starts with a file that already has frontmatter. Then, the `tp.frontmatter` is applied. In other words, `tp.frontmatter` only works if the document already has pre-existing frontmatter to work with.

If you are using this template on a new file, there is no frontmatter to parse because there is no frontmatter yet. In the example case above, you might as well go
```
alias: myfile
note type: seedling
my type: myfile
```

Hope that helps. 
~Welp

**Comment by AB1908** [comment]
This has come up a few times recently. It's worth adding to the docs IMO.

**Comment by welpdx** [comment]
> This has come up a few times recently. It's worth adding to the docs IMO.

Agreed. And I hope @SilentVoid13: 
- adds another trigger (or something) so it is synchronous and
- do something with the cache so that we can get the updated filename, or in this case, the updated frontmatter.

**Comment by hanung665** [comment]
> The documentation starts with a file that already has frontmatter. Then, the `tp.frontmatter` is applied. In other words, `tp.frontmatter` only works if the document already has pre-existing frontmatter to work with.
> 
> If you are using this template on a new file, there is no frontmatter to parse because there is no frontmatter yet. In the example case above, you might as well go
> 
> ```
> alias: myfile
> note type: seedling
> my type: myfile
> ```
> 
> Hope that helps. ~Welp

Thanks. I though this feature is not working in mobile, then I tried on desktop and same thing happen, undefined, it's not working, and I searched and get me to this discussion.

**Comment by ssravjee** [comment]
I have the same issue on Windows 10 Pro with Obsidian v0.15.9

**Comment by jannisgruss** [comment]
I got the same issue while using Templater. In my use case i want to put author and title from my metadata to my heading:

```
---
title: "Titel"
author: "Author"
---

**Autor(en)**: [[<%+ tp.frontmatter.author %>]]
**Titel**: [[<%+ tp.frontmatter.title %>]]
```

I'm getting the String, but it is linked to the template text obviously.
Is there a way to convert the string you get from the tp.frontmatter output again into a useful link?

**Reply by 82Kang** [reply]
Stuck in the same boat! From the above comments, it looks like its the way it is. Had to change my workflow a bit, which I didn't like, by introducing a page property instead of frontmatter.

**Reply by AB1908** [reply]
Are y'all sure you want to insert stuff and not use inline dataview queries instead?
